doc/sphinx_source/mainDocs/tcl-commands.rst

```diff
@@ -164,6 +164,16 @@ cap <active/available/raw> [arg]
 
   Module: server
 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+tagmsg <tag string> <target>
```

Comment:
From a Tcl point of view, I'd prefer an API like
`tagmsg <target> <key> <value> <key2> <value2> ...` or `tagmsg <target> <list of key/value pairs>`

Reply:
This gets tricky depending on the target of the tagmsg. A keypair needs to be prefixed with a '+' if it is intended to be delivered beyond the server (ie, in a PRIVMSG going to a channel), otherwise it is/can be not propagated past the server (ie, the channel doesn't get it). Further, it is possible that some tags on a line would be intended to 'stop' at the server, while others would be intended to continue to other users, allowing for a mixed-use case that may prevent some sort of -flag from being useful. Maybe you could prefix the key in the dict with a +, but I imagine that would negatively affect tcl scripts trying to use it.

Mixed case example:
```
C->S: @label=123;+example-client-tag=example-value TAGMSG #channel
S->Other Clients: @msgid=abc;+example-client-tag=example-value :nick!user@example.com TAGMSG #channel
```

Maybe we could create a helper command that formats a dict into basic tag string formats?
```
set tagdict [dict create a 1 b 2 c 3] 
tagmsg [maketagstring tagdict] #channel 

C->S: @+a=1;+b=2;+c=3 TAGMSG #channel
```
and
```
set tagdict [dict create a 1 b 2 c 3]
tagmsg [maketagstring tagdict -server] #channel 
C->S: @a=1;b=2;c=3 TAGMSG #channel
```
and that allows a scripter to still manually send a mixed-case string?